faunadb-scala/src/main/scala/faunadb/values/time/package.scala

```diff
@@ -0,0 +1,89 @@
+package faunadb.values
+
+import org.joda.time.Instant
+import org.joda.time.format.ISODateTimeFormat
+import scala.Ordered._
+
+package time {
+
+  /**
+    * Represents an instance of [[org.joda.time.Instant]] with nanosecond precision.
+    * It calculates the overflow from nano to milliseconds and adds to the initial instant.
+    */
+  class HighPrecisionTime private(initialInstant: Instant, nanosToAdd: Long) {
```

Comment:
It's implementing Ordering implicitly down as part of the package object. Is that Ok? I found that easier to implement since it relies on another implicit to deal with joda's `Instant` ordering.